airbyte-lib/airbyte_lib/caches/snowflake.py

```diff
@@ -82,6 +82,11 @@ def _write_files_to_new_table(
         """
         return super()._write_files_to_new_table(files, stream_name, batch_id)
 
+    @overrides
+    def _quote_identifier(self, identifier: str) -> str:
+        # TODO: Implement quoting for Snowflake - this is omitted for now because it doesn't play well with case-sensitivity.
```

Comment:
```suggestion
    @overrides
    def _quote_identifier(self, identifier: str) -> str:
        """Apply quoting, currently a no-op for Snowflake.

        TODO: Implement quoting for Snowflake - this is omitted for now because it doesn't play well with case-sensitivity.

        To make this work, we need to set this when creating connection objects.

            # Set QUOTED_IDENTIFIERS_IGNORE_CASE to True
            connection.execute("ALTER SESSION SET QUOTED_IDENTIFIERS_IGNORE_CASE = TRUE")
        """
```